clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/CoreNotificationRenderer.java

```diff
@@ -51,7 +48,7 @@ public String getTitle(final Bundle extras, final Context context) {
     }
 
     @Override
-    public NotificationCompat.Builder renderNotification(final Bundle extras, final Context context,
+    public Builder renderNotification(final Bundle extras, final Context context,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 113-125 into a separate method.